acp/DPI.md

```diff
@@ -0,0 +1,45 @@
+## DefraDB Policy Interface (DPI) Restrictions
+To be a DPI compliant resource, these rules MUST be met:
+- The resource must exist.
+- The resource must atleast have the required registerer relation under `relations`.
+- The resource must atleast have all the required permissions under `permissions`.
+- Every required permission must have the required registerer relation in `expr`.
+- The required registerer relation must be the leading (first) relation in `expr` (see example below).
+
+To be a 'DPI' policy, all of it's resources must be DPI compliant.
+To be 'Partially-DPI' atleast one of it's resource must be DPI compliant.
+
+Learn more about DefraDB ACP Terminologies [TERMINOLOGY](/acp/TERMINOLOGY.md)
+
+### Moreover:
+All the required permissions are:
+    - Defined in the `dpi.go` file in variable `dpiRequiredPermissions`.
+The required registerer relation name is:
+    - Defined in the `dpi.go` file in variable `requiredRegistererRelationName`.
+
+## DPI Resource Examples:
+- Check out tests here: [tests/integration/acp/schema/add_dpi](/tests/integration/acp/schema/add_dpi)
+- The tests linked are broken into `accept_*_test.go` and `reject_*_test.go` files.
+- Accepted tests document the valid DPIs (as the schema is accepted).
+- Rejected tests document invalid DPIs (as the schema is rejected).
+- There are also some Partially-DPI tests that are both accepted and rejected depending on the resource.
+
+### DPI Resource's Required Permission's Expression:
+Eventhough the following expressions are valid generic policy expressions, they will make a
+DPI compliant resource loose it's DPI status as these expressions are not in accordance to
```

Comment:
todo: `loose` => `lose`, `it's` => `its` (I think :) maybe double check 'its' lol)

Reply:
- [ ] fix typo lose